### PR TITLE
node pools: add `node_pool` field to job spec

### PR DESCRIPTION
**Pull request by tgross**
This changeset only adds the `node_pool` field to the jobspec, and ensures that it gets picked up correctly as a change. Without the rest of the implementation landed yet, the field will be ignored. Includes required updates to the docs and `nomad job status` command.

### CODE REVIEW

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -163,6 +163,9 @@ const (
 	DefaultNamespace            = "default"
 	DefaultNamespaceDescription = "Default shared namespace"
 
+	// DefaultNamespace is the default node pool.
+	DefaultNodePool = "default"
```
Comment (lgfa29) [inline nomad/structs/structs.go:167]: I have this in `structs_node_pool.go` but with the name flipped 😅 
https://github.com/hashicorp/nomad/blob/45fcc59a82137bf69cb1f6087e5ce614c378cd34/nomad/structs/node_pool.go#L20-L23

File: website/content/api-docs/json-jobs.mdx [hunk 2]
```diff
@@ -256,6 +257,8 @@ The `Job` object supports the following keys:
 - `Datacenters` - A list of datacenters in the region which are eligible
   for task placement. This must be provided, and does not have a default.
 
+- `NodePool` - The node pool in which the job can be placed. Defaults to "default".
```
Comment (lgfa29) [inline website/content/api-docs/json-jobs.mdx:260]:
```suggestion
- `NodePool` - The node pool in which the job can be placed. Defaults to `"default"`.
```

**Review by lgfa29 — APPROVED** [review]
I have those constants somewhere in my commit log, but I can clean them up once I get to them.

File: nomad/structs/structs.go (hunk 1 above)
Comment (tgross) [reply nomad/structs/structs.go:167]: Hrm, we have it as `DefaultNodePool` in `api` in your [private branch](https://github.com/hashicorp/nomad-enterprise/blob/f-node-pools-nodes-api/api/node_pools.go#L8) which is why I did that here. It also matches `DefaultNamespace`. We should probably be consistent between `api` and `nomad/structs` just so we don't frustrate ourselves. Which would you prefer?

Comment (tgross) [reply nomad/structs/structs.go:167]: Eh, because there's also a `NodePoolAll` it makes for nicer autocomplete to have it be `NodePoolDefault`. Will fix.

Comment (lgfa29) [reply nomad/structs/structs.go]: Ah sorry, I didn't notice the inconsistency. Yeah, I think `NodePoolDefault` and `NodePoolAll` work better. I reconciled their use in https://github.com/hashicorp/nomad/pull/17388.

**Pull request merged**